### PR TITLE
January 2023 newsletter - ready for review

### PR DESCRIPTION
**Pull request by benjaoming**
@humitos I added a quick copy of the big "call to action" button that you so ingeniously suggested :)

Now a question will remain... what action should we choose? We don't have to have the same action each time. We can even have several.

### CODE REVIEW

**Comment by humitos** [comment]
> Now a question will remain... what action should we choose? We don't have to have the same action each time. We can even have several.

Yeah, exactly. The action will depend on each newsletter --it has to be something related to the actual newsletter, not something generic. So, if the most important thing in the newsletter is "Custom domains are now a lot easier", we can link to the documentation page for custom domains with "Add a custom domain to your docs now"

**Comment by benjaoming** [comment]
Sounds good :+1: We can see what works once we have the content for this edition at hand.

File: newsletter-january-2023.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+.. Here are the latest updates from our team:
+
+- 📹️ Eric's talk at DjangoCon US can be watched here: `Documenting Django Code in 2022`_
+- 🛠️ We are busy reorganizing our own documentation. Since last newsletter, we have opened or merged 20 documentation PRs. We look forward to a "show and tell" about how all of this was refactored using Diátaxis framework.
+- 🚢️ Sphinx 6.0 and 6.1 have been released. Read :doc:`our considerations for upgrading </sphinx6-upgrade>`.
+- ⚙️ Add new stuff here
+- 💡️ Choose your own emoji
+- 🔍️ 404 pages are being improved by contextualization the user message, giving relevant guidance to readers and project owners.
+
+You can always see the latest changes to our platforms in our :doc:`Read the Docs Changelog <readthedocs:changelog>`.
```
Comment (stsewd) [inline newsletter-january-2023.rst:30]: We should probably mention https://github.com/readthedocs/readthedocs.org/security/advisories/GHSA-368m-86q9-m99w

File: newsletter-january-2023.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+=======================================
+
+Happy 2023! In this newsletter, we present the updates happening in 2022's last month and since the :doc:`previous newsletter </newsletter-december-2022>`.
+
+.. Commenting out this stuff, isn't it better that 
+.. 
+.. News and updates
+.. ----------------
+.. 
+.. Here are the latest updates from our team:
```
Comment (humitos) [inline newsletter-january-2023.rst:21]: I think this section gives some context to the following bullets. Otherwise, I'm not sure it feels fine to start reading a bunch of items without too much context about what they are.

File: newsletter-january-2023.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+- 📹️ Eric's talk at DjangoCon US can be watched here: `Documenting Django Code in 2022`_
+- 🛠️ We are busy reorganizing our own documentation. Since last newsletter, we have opened or merged 20 documentation PRs. We look forward to a "show and tell" about how all of this was refactored using the `Diátaxis Framework`_.
+- 🚢️ Sphinx 6.0 and 6.1 have been released. Read :doc:`our considerations for upgrading </sphinx6-upgrade>`.
+- ⚙️ Add new stuff here
+- 💡️ Choose your own emoji
+- 🔍️ 404 pages are being improved by contextualization the user message, giving relevant guidance to readers and project owners.
```
Comment (humitos) [inline newsletter-january-2023.rst:28]: Didn't we already talked about this in a previous newsletter?

File: newsletter-january-2023.rst [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+Happy 2023! In this newsletter, we present the updates happening in 2022's last month and since the :doc:`previous newsletter </newsletter-december-2022>`.
+
+.. Commenting out this stuff, isn't it better that 
+.. 
+.. News and updates
+.. ----------------
+.. 
+.. Here are the latest updates from our team:
+
+- 📹️ Eric's talk at DjangoCon US can be watched here: `Documenting Django Code in 2022`_
```
Comment (humitos) [inline newsletter-january-2023.rst:23]:
```suggestion
- 📹️ Eric delivered a talk at DjangoCon US 2022 about the state of art to document a project in 2022. You can watch it here: `Documenting Django Code in 2022`_
```

File: newsletter-january-2023.rst [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+- 🛠️ We are busy reorganizing our own documentation. Since last newsletter, we have opened or merged 20 documentation PRs. We look forward to a "show and tell" about how all of this was refactored using the `Diátaxis Framework`_.
```
Comment (humitos) [inline newsletter-january-2023.rst:24]:
```suggestion
- 🛠️ We continued reorganizing our own documentation to follow `Diátaxis Framework`_.
  We look forward to a "show and tell" about our experience refactoring them using this framework.
```

File: newsletter-january-2023.rst [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+- A search redesign to make it nicer across our dashboard and in-doc search experiences. 
+- 
+
+Possible issues
+---------------
+
+Sphinx 6 and 6.1 were released very recently and you may experience issues if you upgrade or if you have not pinned your documentation dependencies.
+
+We are actively updating our theme to support Sphinx and docutils updates. If you find regressions in any new releases of the `sphinx-rtd-theme <https://sphinx-rtd-theme.readthedocs.io/>`_,
+please don't hesitate to `open an issue on GitHub <https://github.com/readthedocs/sphinx_rtd_theme/>`_.
```
Comment (humitos) [inline newsletter-january-2023.rst:49]: Please, follow semantic line break all across the document.

File: newsletter-january-2023.rst (hunk 3 above)
Comment (humitos) [reply newsletter-january-2023.rst:28]: It seems we have the same line in the December newsletter: https://blog.readthedocs.com/newsletter-december-2022/

BTW, I don't like this patter of copy&paste the previous one without removing the old content. Iit's hard to know what's really going out in _this one in particular_. If it's a placeholder, it should be clear it's a placeholder. Otherwise, it's confusing for reviewers.

File: newsletter-january-2023.rst [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+.. _Documenting Django Code in 2022: https://www.youtube.com/watch?v=mqn0D4xat58
+.. _Diátaxis Framework: https://diataxis.fr/
+
+Upcoming features
+-----------------
+
+- We're working on improving our integration with `Material for MkDocs <https://squidfunk.github.io/mkdocs-material/>`_, which is a great theme for `MkDocs <https://www.mkdocs.org/>`_ documentation projects.
+- Many improvements to our URL handling code, which will allow us to support more flexible URL configurations for projects.
+- A search redesign to make it nicer across our dashboard and in-doc search experiences.
```
Comment (humitos) [inline newsletter-january-2023.rst:40]: Here as well. This is old content from the December newsletter.

File: newsletter-january-2023.rst [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+- We're working on improving our integration with `Material for MkDocs <https://squidfunk.github.io/mkdocs-material/>`_, which is a great theme for `MkDocs <https://www.mkdocs.org/>`_ documentation projects.
+- Many improvements to our URL handling code, which will allow us to support more flexible URL configurations for projects.
+- A search redesign to make it nicer across our dashboard and in-doc search experiences. 
+- 
+
+Possible issues
+---------------
+
+Sphinx 6 and 6.1 were released very recently and you may experience issues if you upgrade or if you have not pinned your documentation dependencies.
```
Comment (humitos) [inline newsletter-january-2023.rst:46]:
```suggestion
Sphinx 6.0 and 6.1 were released very recently and you may experience issues if you upgrade or if you have not pinned your documentation dependencies.
```

I think we should tell the users what to do here or how to fix the issue or something. Some guidance in case they hit an issue.

I don't think we have to explain _the solutions_ here, but at least a link somewhere to read more --or how to pin the dependencies, etc.

File: newsletter-january-2023.rst [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+Upcoming features
+-----------------
+
+- We're working on improving our integration with `Material for MkDocs <https://squidfunk.github.io/mkdocs-material/>`_, which is a great theme for `MkDocs <https://www.mkdocs.org/>`_ documentation projects.
+- Many improvements to our URL handling code, which will allow us to support more flexible URL configurations for projects.
+- A search redesign to make it nicer across our dashboard and in-doc search experiences. 
+- 
+
+Possible issues
+---------------
```
Comment (humitos) [inline newsletter-january-2023.rst:44]: The two issues mentioned here, should probably a bullet list since they are two different things. It's easier to skim and read in a list.

**Review by humitos — COMMENTED** [review]
I think this is a good start!

I mentioned this already in an inline comment, but I wanted to highlight it globally as well. I think we shouldn't copy&paste the content of the previous newsletter and just edit it _without removing the old content_. I reviewed the full document and in multiple opportunities I thought "Weird, I think I've read this before" and there were some highlights/features that were already shipped in the previous newsletter.

I'd prefer if we start with a clean structure of the document without old content that's not going out in this newsletter to avoid these situations.

File: newsletter-january-2023.rst (hunk 2 above)
Comment (benjaoming) [reply newsletter-january-2023.rst:21]: Aren't you commenting on something that I commented _out_ (because of the same reasons you also stated) ? :)

File: newsletter-january-2023.rst (hunk 3 above)
Comment (benjaoming) [reply newsletter-january-2023.rst:29]: Sure, like this is not open "for review" but a draft where you can add ideas.

File: newsletter-january-2023.rst (hunk 7 above)
Comment (benjaoming) [reply newsletter-january-2023.rst:41]: I know

File: newsletter-january-2023.rst (hunk 6 above)
Comment (benjaoming) [reply newsletter-january-2023.rst]: I hope you know that the intention is to do that? We write an awful lot of text every day _without_ semantic line breaks, so it's hard to practice a natural habit with this.

File: newsletter-january-2023.rst (hunk 1 above)
Comment (benjaoming) [reply newsletter-january-2023.rst:32]: Added :+1: 

I hope it's fine to add with no further context, I feel like it's hard to give a quick summary and that only people that self-host really need to review it.

File: newsletter-january-2023.rst (hunk 2 above)
Comment (humitos) [reply newsletter-january-2023.rst:23]: It seems you commented it to remove it. My point is that we should keep it because it works fine.

Comment (benjaoming) [reply newsletter-january-2023.rst:23]: Sure, I can bring it back. I wanted to remove it to get the list further up on the screen.

Writing "Newsletter", then an intro, then "News and updates", then "Here are the latest updates from our team" just seemed like a slow start. Thought it'd be more simple to present the list as quickly as possible.

File: newsletter-january-2023.rst (hunk 7 above)
Comment (benjaoming) [reply newsletter-january-2023.rst:43]: "Upcoming features" has had duplications previously. I would like to keep it because it ensures that there is a follow up. I.e. that we don't announce an upcoming feature and then ghost it in the next newsletter.

Comment (humitos) [reply newsletter-january-2023.rst:43]: I would avoid publishing duplicated content on the newsletters. Seems like a bad practice to me and it's like reading old news.

Publishing that "we are working on something" it does not mean that it has to be completed before the next newsletter. It also does not mean that if we don't finish it, we will repeat the announcement.

I'm fine with a "follow-up", but we should give _something new/extra_ to our readers --not the same content. If we made some progress on that task, we can tell them about that progress.

Comment (benjaoming) [reply newsletter-january-2023.rst:43]: > I would avoid publishing duplicated content on the newsletters. Seems like a bad practice to me and it's like reading old news.

I agree with this, I don't want that neither. As with previous newsletters, I'll go over these items and see if they will have a "news update" or if there's another way to keep readers updated on the progress.

I can add a Sphinx comment to clarify that the intention isn't to repeat the text.

File: newsletter-january-2023.rst (hunk 2 above)
Comment (benjaoming) [reply newsletter-january-2023.rst]: I re-added it so we can get the newsletter out. I still don't like to have such a long intro, but I'll repeat the proposal in the next newsletter.

File: newsletter-january-2023.rst (hunk 7 above)
Comment (benjaoming) [reply newsletter-january-2023.rst]: Added some notes as a comment.

File: newsletter-january-2023.rst [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+Read the Docs for Business
+--------------------------
+
+Considering using Read the Docs for an organization?
+
+.. raw:: html
+
+   <a href="https://about.readthedocs.com/" style="background-color: #409cff; border-radius: 3px; color: #ffffff; display: block; margin: 30px auto; font-size: 18px; font-weight: 700; line-height: 24px; padding: 15px 0 15px 0; text-align: center; text-decoration: none; width: 238px;">
+     Discover all the features ✨️
+   </a>
```
Comment (benjaoming) [inline newsletter-january-2023.rst:110]: Any feedback on this specific proposal or ideas for a different "call to action" button?

@humitos suggested at the last newsletter that we start having a large button. I didn't find anything in the content to immediately connect the message to, so this is a bit generic.

File: newsletter-january-2023.rst [hunk 11]
```diff
@@ -0,0 +1,118 @@
+.. post:: Jan 1, 2023
```
Comment (benjaoming) [inline newsletter-january-2023.rst:1]:
```suggestion
.. post:: Jan 6, 2023
```

File: newsletter-january-2023.rst [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+The core team has laid out its 2023 Q1 roadmap and our general focus will be on:
+
+- Improving support for using *any* documentation tool in the build process.
+- Currently, users can either customize a default build process or overwrite the entire process from scratch.
+  We are aligning these two approaches,
+  making it easier to design and configure builds for a project's individual needs.
+- Continuing to restructure and rewrite the user documentation of Read the Docs.
+  We look forward to publishing the restructured documentation which is likely happening at the end of Q1.
+- Dashboard redesign: The entire Dashboard will be redesigned and re-implemented.
+  This work is about to be launched!
```
Comment (benjaoming) [inline newsletter-january-2023.rst:64]: These are not very specific "features". I ended up using high-level roadmap items instead of issues/PRs, but it seems good since we're kicking off a new quarter?

I did re-formulate things a lot, the intention is to put focus on relevance to users. Not sure if I succeed very well in that.

File: sphinx6-upgrade.rst [hunk 13]
```diff
@@ -34,6 +34,26 @@ we recommend checking for Sphinx 6 support in those projects before upgrading.
...
+    Visiting the issue trackers of these projects is helpful.
+
+* We are actively updating our theme to support Sphinx and docutils updates.
+  If you find regressions in any new releases of the `sphinx-rtd-theme <https://sphinx-rtd-theme.readthedocs.io/>`_,
+  please don't hesitate to `open an issue on GitHub <https://github.com/readthedocs/sphinx_rtd_theme/>`_.
+
+If you suspect that upgrade issues are caused by dependency mismatches,
+we recommend taking the approach of *reproducible builds*.
+This includes explicitly specifying all relevant dependencies.
+Read more in our documentation's :doc:`introduction to Reproducible Builds <readthedocs:guides/reproducible-builds>`.
```
Comment (ericholscher) [inline sphinx6-upgrade.rst:54]: This is definitely useful 👍

**Review by ericholscher — APPROVED** [review]
This looks good overall, with a few nits 👍

File: newsletter-january-2023.rst (hunk 10 above)
Comment (benjaoming) [reply newsletter-january-2023.rst:110]: Great! I'll make a note for the next newsletter to consider more "call to action" ideas.

File: sphinx6-upgrade.rst [hunk 14]
```diff
@@ -34,6 +34,26 @@ we recommend checking for Sphinx 6 support in those projects before upgrading.
...
 
+Troubleshooting
+---------------
+
+* Sphinx 6.0 and 6.1 are very fresh releases. As such, please remember:
+
+  * Not all extensions and themes have added support for the new versions!
+  * You may experience issues if you have not pinned your documentation dependencies.
+  * Issues may also be caused by new bugs in the Sphinx 6 compatibility of extensions and themes.
+    Visiting the issue trackers of these projects is helpful.
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply sphinx6-upgrade.rst:45]: I renamed it from "Troubleshooting" to "Possible issues", I think the section makes sense now.

> This seems like it's just rehashing the content above? Not sure what value it's adding.

It rehashes contents of the Newsletter but not from the Sphinx 6 Upgrade post AFAICT.

[the remark this answers is not in the document]
Comment (benjaoming) [reply sphinx6-upgrade.rst:59]: > I'd probably have a few suggestions for ways to work around it (presumably an explicit syntax for pinning old Sphinx?).

That sounds like a good idea for the Newsletter's list of Possible Issues.

I added a section in the Sphinx 6 upgrade considerations called "Don't fancy upgrading?"

File: newsletter-january-2023.rst (hunk 12 above)
[the remark this answers is not in the document]
Comment (humitos) [reply newsletter-january-2023.rst]: Yeah, I don't know if makes sense mentioning it _again_. It seems like "old news" to me. We've been promoting this work since April 2021: https://github.com/readthedocs/blog/blob/d627e9fc6e183ff40c8f152375ec42c563daaa81/newsletter-april-2021.rst?plain=1#L91-L94

If we want to mention this, we should probably rephrase it to say that we are going to resume working on this or similar.

[the remark this answers is not in the document]
Comment (benjaoming) [reply newsletter-january-2023.rst]: It's funny though how @ericholscher's formulation is so consistent with almost 2 years span - "we hope to launch in public beta soon" and "we are hoping to have a public beta in Q1"

It does seem like a good thing to know what the difference is "this time" from the previous time that it was mentioned. I imagine that we have become a lot better at planning? So perhaps there is a bit more "confidence" behind this announcement.

I'll remove the beta announcement. I'll just write that we are working towards a public beta and not anything about when it's launched?

**Comment by ericholscher** [comment]
Should we note the CircleCI breach as well? That seems notable, or should we do a security advisory for it?

File: sphinx6-upgrade.rst [hunk 15]
```diff
@@ -34,6 +34,26 @@ we recommend checking for Sphinx 6 support in those projects before upgrading.
 .. _sphinxcontrib-jquery: https://pypi.org/project/sphinxcontrib.jquery/
 
 
+Troubleshooting
+---------------
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply sphinx6-upgrade.rst]: Yes, agreed, there were some synergies between the two posts, but I can see that it's definitely annoying to review. Sry :/

File: newsletter-january-2023.rst [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+  We are aligning these two approaches,
+  making it easier to design and configure builds for a project's individual needs.
+- Continuing to restructure and rewrite the user documentation of Read the Docs.
+  We look forward to publishing the restructured documentation which is likely happening at the end of Q1.
+- Dashboard redesign: Our admin dashboard is currently being redesigned and reimplemented.
+  We are hoping to have a public beta in Q1.
+
+Interested in all the details? `View our full Roadmap 📍️`_
+
+.. _View our full Roadmap 📍️: https://github.com/orgs/readthedocs/projects/156/views/4
```
Comment (benjaoming) [inline newsletter-january-2023.rst:77]: @ericholscher pay attention to this one! @humitos and I was on a call and he suggested that we add a button with the roadmap since we are talking a lot about it. We then looked at a previous mention of the roadmap, where it was said that we would announce it in the future.

We agreed to do a soft announcement and then perhaps write something more in the next newsletter, so we can brew up at text that tells readers what the intention is with this link -- like why should you bother, how do you tell us that you want a bug fix or a feature etc.

Comment (ericholscher) [reply newsletter-january-2023.rst:77]: Yea, makes sense 👍

**Review by ericholscher — APPROVED** [review]
